### PR TITLE
Add documentation for CC CORS

### PR DESCRIPTION
**Pull request by kyguy**
Signed-off-by: Kyle Liberti <kliberti@redhat.com>

### Type of change

- Documentation

### Description

Documentation for https://github.com/strimzi/strimzi-kafka-operator/pull/3816

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — APPROVED** [review]

File: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc [hunk 1]
```diff
@@ -40,6 +40,30 @@ spec:
     # ...
 ----
 
+[[cors-configuration]]
+== Cross-origin resource sharing configuration
```
Comment (ppatierno) [inline documentation/modules/cruise-control/ref-cruise-control-configuration.adoc:44]: I would use capital letters for making CORS ... "Cross-Origin Resource Sharing configuration"

File: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc [hunk 2]
```diff
@@ -40,6 +40,30 @@ spec:
...
+
+By default, cross-origin resource sharing (CORS) for Cruise Control is disabled and can only be enabled for read-only access.
```
Comment (ppatierno) [inline documentation/modules/cruise-control/ref-cruise-control-configuration.adoc:46]: ditto as above

**Review by ppatierno — APPROVED** [review]
LGTM. Just a nit.

**Comment by laidan6000** [comment]
Remember to update the changelog.

File: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc [hunk 3]
```diff
@@ -40,6 +40,30 @@ spec:
     # ...
 ----
 
+[[cors-configuration]]
```
Comment (laidan6000) [inline documentation/modules/cruise-control/ref-cruise-control-configuration.adoc:43]: Add `[discrete]` after line 43.

File: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc [hunk 4]
```diff
@@ -40,6 +40,30 @@ spec:
...
+== Cross-Origin Resource Sharing configuration
+
```
Comment (laidan6000) [inline documentation/modules/cruise-control/ref-cruise-control-configuration.adoc:45]: _Add a general definition of CORS in a separate paragraph:_

Cross-Origin Resource Sharing (CORS) allows you to specify allowed methods and originating URLs for accessing REST APIs.

File: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc [hunk 5]
```diff
@@ -40,6 +40,30 @@ spec:
...
+By default, Cross-Origin Resource Sharing (CORS) for Cruise Control is disabled and can only be enabled for read-only access.
```
Comment (laidan6000) [inline documentation/modules/cruise-control/ref-cruise-control-configuration.adoc:46]:
```suggestion
By default, CORS is disabled for the Cruise Control REST API. 
When enabled, only `GET` requests for read-only access to the Kafka cluster state are allowed. 
```

File: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc [hunk 6]
```diff
@@ -40,6 +40,30 @@ spec:
...
+This means that applications running in origins different from Strimzi components cannot send POST requests to the Cruise Control API.
```
Comment (laidan6000) [inline documentation/modules/cruise-control/ref-cruise-control-configuration.adoc:47]:
```suggestion
This means that external applications, which are running in different origins than the Strimzi components, cannot make `POST` requests to the Cruise Control API. 
```

File: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc [hunk 7]
```diff
@@ -40,6 +40,30 @@ spec:
     # ...
 ----
 
+[[cors-configuration]]
+== Cross-Origin Resource Sharing configuration
+
+By default, Cross-Origin Resource Sharing (CORS) for Cruise Control is disabled and can only be enabled for read-only access.
+This means that applications running in origins different from Strimzi components cannot send POST requests to the Cruise Control API. 
+However, these same applications can still send GET requests to the Cruise Control API when cors is enabled for Cruise Control.
```
Comment (laidan6000) [inline documentation/modules/cruise-control/ref-cruise-control-configuration.adoc:48]:
```suggestion
However, those applications can make GET requests to access read-only information about the Kafka cluster, such as the current cluster load or the most recent optimization proposal. 
```

File: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc [hunk 8]
```diff
@@ -40,6 +40,30 @@ spec:
...
+This prevents third-party applications from manipulating Kafka cluster state through Cruise Control without Strimzi approval but still allows third-party applications to access read-only cluster information from Cruise Control.
```
Comment (laidan6000) [inline documentation/modules/cruise-control/ref-cruise-control-configuration.adoc:49]: Do we need to say anything about how Strimzi "approves" a CORS request? It feels like some information is missing here.

If nothing else is needed, then I think we can lose the sentence (line 49).

File: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc [hunk 9]
```diff
@@ -40,6 +40,30 @@ spec:
...
+
```
Comment (laidan6000) [inline documentation/modules/cruise-control/ref-cruise-control-configuration.adoc:50]: _Add a sub-heading and a new sentence:_

.Enabling CORS for Cruise Control

You enable CORS in `Kafka.spec.cruiseControl.config`.

File: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc [hunk 10]
```diff
@@ -40,6 +40,30 @@ spec:
...
+.An example of enabling Cruise Control CORS
```
Comment (laidan6000) [inline documentation/modules/cruise-control/ref-cruise-control-configuration.adoc:51]: I think line 51 can be deleted.

File: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc [hunk 11]
```diff
@@ -40,6 +40,30 @@ spec:
...
+  name: my-cluster
+spec:
+  # ...
+  cruiseControl:
+    # ...
+    config:
+      webserver.http.cors.enabled: true
+    # ...
+----
+
```
Comment (laidan6000) [inline documentation/modules/cruise-control/ref-cruise-control-configuration.adoc:66]: _Add a link to the Wiki:_

For more information, see [REST APIs](https://github.com/linkedin/cruise-control/wiki/REST-APIs) in the Cruise Control Wiki.

File: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc [hunk 12]
```diff
@@ -40,6 +40,30 @@ spec:
...
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  # ...
+  cruiseControl:
+    # ...
+    config:
```
Comment (laidan6000) [inline documentation/modules/cruise-control/ref-cruise-control-configuration.adoc:62]: Are there any more configuration options for CORS? For example, how do you configure the allowed origins, like you can for the Kafka Bridge?

**Review by laidan6000 — CHANGES_REQUESTED** [review]
Peer reviewed.

File: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc [hunk 13]
```diff
@@ -40,6 +38,39 @@ spec:
...
 
+[[cors-configuration]]
+[discrete]
+== Cross-Origin Resource Sharing configuration
+
+Cross-Origin Resource Sharing (CORS) allows you to specify allowed methods and originating URLs for accessing REST APIs.
+
+By default, CORS is disabled for the Cruise Control REST API.
+When enabled, only `GET` requests for read-only access to the Kafka cluster state are allowed.
+his means that external applications, which are running in different origins than the Strimzi components, cannot make `POST` requests to the Cruise Control API.
```
Comment (laidan6000) [inline documentation/modules/cruise-control/ref-cruise-control-configuration.adoc:49]: Typo line 49: his.

File: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc [hunk 14]
```diff
@@ -40,6 +38,39 @@ spec:
...
+However, those applications can make GET requests to access read-only information about the Kafka cluster, such as the current cluster load or the most recent optimization proposal.
```
Comment (laidan6000) [inline documentation/modules/cruise-control/ref-cruise-control-configuration.adoc:50]:
```suggestion
However, those applications can make `GET` requests to access read-only information about the Kafka cluster, such as the current cluster load or the most recent optimization proposal.
```

File: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc [hunk 15]
```diff
@@ -40,6 +38,39 @@ spec:
...
+
+.Enabling CORS for Cruise Control
+
+You enable CORS in `Kafka.spec.cruiseControl.config`.
```
Comment (laidan6000) [inline documentation/modules/cruise-control/ref-cruise-control-configuration.adoc:54]:
```suggestion
You enable and configure CORS in `Kafka.spec.cruiseControl.config`. 
For example:
```

**Review by laidan6000 — CHANGES_REQUESTED** [review]
Thanks for making the changes. I've left a few more comments.

**Review by laidan6000 — APPROVED** [review]
Changes look good.

**Comment by laidan6000** [comment]
Small typo: "You enable ad configure"

**Comment by kyguy** [comment]
Sorry @laidan6000, I keep missing these typos!

**Comment by scholzj** [comment]
Thanks @kyguy

**Pull request merged**